Scope revisions

noamross opened the pull request:
This is a revision of our scope guidelines, primarily clarifying language and providing examples. However, it includes some substantive changes, some of which are large but meant as starting points for conversation among @ropensci/editors. I hope to use this PR to facilitate this discussion and then resolve it in not too much time.

Substantive changes:

-   *drop the data visualization category*: This basically opens up to any htmlwidget package, and I don't see how to to deal with it it a reasonable way. It seems to have been there primarily for the legacy of **plotly**.
-   *"reproducibility" includes citation tools*: I believe **RefManageR** would fall under data munging, but citation is a core functionality for reproducibility that is specific to scientific workflows, and I think we should explicitly include it. Similarly I think **knitcitations** would fit into this mission.
-   *"reproducibility" does not include general R markdown tools*: I've tried to make a reasonable definition here, as this category can be very broad.
-   Explicit description of the text analysis pilot.
-   *add an explicit category for database wrappers*: We have a lot of these developed internally.

I think its fine that some legacy and current in-process packages would fall out of scope under changes, and I note this in the text.

This is kicked off from the useful comment from @statsmaths (please chime in with your thoughts on these changes - do they help?). Tagging @lmullen regarding the text analysis component.

noamross commented:
Should I rename the section "Aims and Scope" as that parlance is familiar to journal people?

maelle commented:
Thanks! :ok_hand: Regarding the general RMarkdown sentence I was first puzzled because of https://github.com/ropensci/ezknitr but then your last sentence about old packages clarifies it IMO.

Yes for renaming.

lmullen commented:
I'm very glad that rOpenSci is willing to pilot a text analysis sub-speciality, and that the process is being formalized. 

My one suggestion pertains to a distinction that @noamross and I and perhaps others discussed on Slack. The idea was that text analysis packages could implement but not create new statistical methods. For example, wordVectors fits because it implements word2vec and can cite the Mikolov et al. paper. So rOpenSci is only undertaking the peer review of the package and not of word embedded models altogether.

Would that distinction be worth making in this text? Perhaps matters are sufficiently undecided that that distinction would be to precise.

kbenoit commented:
I'm swamped with exams but just caught this thread. Where do I look to catch up with the subspecialty for text analysis packages? I'm very interested in this intriguing idea. Would be nice to be able to install the set using e.g. the tidyverse approach. The key will be in setting standards. Where is this discussion happening?

noamross commented:
Whoops, meant to tag @statsmaths, not statsmath

noamross commented:
@kbenoit See the `onboarding` Slack channel conversation starting January 25.  @lmullen Yes, I think that's correct. Will update to make this clear.

sckott commented on policies.md:
```diff
@@ -7,28 +7,84 @@
+-  Exploratory data analysis packages that visualize or summarize data. 
+-  General workflow or package development support packages
+
+We encourage submitting packages not accepted to rOpenSci those directly to CRAN,
```
"We encourage submitting packages not accepted to rOpenSci directly to CRAN" ?

sckott commented on policies.md:
```diff
@@ -7,28 +7,84 @@
+
+Note that not all packages developed internally by rOpenSci or through our events
+or collaborations are in-scope for onboarding process.  As our scope has evolved
+over time, some previously packages would no longer be in-scope.
```
"some packages we previously accepted would no longer be in-scope." ??

sckott commented on policies.md:
```diff
@@ -7,28 +7,84 @@
+[**robotstxt**](https://github.com/ropensci/onboarding/issues/25),
+[**genbankr**](https://github.com/ropensci/onboarding/issues/47))
+
+* __database access__: Bindings and wrappers generic databse APIs
```
"Bindings and wrappers for generic databse APIs"

sckott commented:
"Aims and scope" is good

I think we should state that:

- we may make exceptions under certain scenarios - though we strive to be consistent
- our policies/categories are a living document and change through time 

How do we think about the categories in `policies.md` jive with the categories used on the packages page https://ropensci.org/packages/ ?

noamross commented:
Fixed typos @sckott found, addressed @sckott and @lmullen's comment.

These don't line up super well with the web site.  I'm not that concerned about this, as categorization for discovery doesn't serve the same role.  But if we get to serving that page via the RO API we might have package tags of some sort, which we can implement as tags in this repo. (`topic:` tags v `category:` tags, maybe)

statsmaths commented:
I think these changes go a long way to making the requirements much more clear. Or, at least, it is clear that there requirements for text analysis packages are still very much a work in progress. 

On a related note, the website also has a section for image analysis. Is there plans to add a track for image processing as well? If there is a place where that discussion is happening I'd very much like to be involved.

noamross commented:
@statsmaths There's not at the moment but we'll keep you in mind.  One thing we need to do is do a better job of distinguishing stuff developed in-house with stuff that's in-scope for onboarding. There's not 100% overlap currently.

sckott commented:
True, i guess we don't need onboarding categories to match up with those on `/packages` page

sckott commented:
> One thing we need to do is do a better job of distinguishing stuff developed in-house with stuff that's in-scope for onboarding. There's not 100% overlap currently.

we could have a badge on packages page (and elsewhere) that a pkg is community vs. staff (though some pkgs may not fit either of those perfectly if it’s a collaboration between staff and community)

noamross commented:
I think what we need is not so much community v staff but peer-reviewed or not.  But before we get too far off topic, any other changes to the scope update, (or the EiC responsibilities?).  @karthik, haven't heard from you yet.  No opinions on dropping data viz or including citation tools?

karthik commented:
@noamross Sorry, just getting around to reviewing this and the PR is merged. Thankfully this all makes sense to me. I also think that we need to have just one badge for peer-reviewed (and also make some policy/exception to grandfather all of rOpenSci's existing work that pre-dates onboarding). 

**Suggestions:**

> Reviewers: @githubname1 @githubname2 Due date: YYYY-MM-DD

Change that to 1 reviewer per line. That way if they are not starting the same day (e.g. you've assigned one reviewing but still seeking another), you can have separate due dates.

njtierney commented:
Can I chime in about data visualisations?

There could still be scope for useful data visualisation tools that could be practical to specific fields - say for example certain extensions to ggplot, plotly, or leaflet, that might be very useful for weather plotting, or ecology. Recently, Carson put up plotdap for example.

Perhaps there could be a caveat that describes how the data visualisations need to be applied to specific examples? This is a tricky topic, I think!
